Packages/com.unity.inputsystem/InputSystem/InputSystem.cs

```diff
@@ -3010,114 +3010,122 @@ internal static bool ShouldDrawWarningIconForBinding(string bindingPath)
         #region Actions
 
 #if UNITY_INPUT_SYSTEM_PROJECT_WIDE_ACTIONS
-
-#if UNITY_EDITOR
-        [InitializeOnLoad]
-        public static class PlayStateNotifier
+        // EnteredEditMode  Occurs during the next update of the Editor application if it is in edit mode and was previously in play mode.
+        // ExitingEditMode  Occurs when exiting edit mode, before the Editor is in play mode.
+        // EnteredPlayMode  Occurs during the next update of the Editor application if it is in play mode and was previously in edit mode.
+        // ExitingPlayMode  Occurs when exiting play mode, before the Editor is in edit mode.
+        //
+        // Using the EnteredEditMode / EnteredPlayMode states to transition the actions' enabled
+        // state ensures that the they are active in all of these MonoBehavior methods:
+        //
+        //      Awake() /  Start() / OnEnable() / OnDisable() / OnDestroy()
+        //
+        private static void EnableActions()
         {
-            static PlayStateNotifier()
-            {
-                EditorApplication.playModeStateChanged += ModeChanged;
-            }
+#if UNITY_EDITOR
+            // Abort if not in play-mode in editor
+            if (!EditorApplication.isPlaying)
```

Comment:
Changing this to

`            if (!EditorApplication.isPlayingOrWillChangePlaymode)
`
... leads to the actions being enabled in all scripting states that I checked (OnEnable / Awake / Start &.)

Reply:
This is interesting, I would suggest we take it as something to consider on top of this (isolated) instead of messing this up more than it already is. The behavior between player and play-mode on this branch doesn't change this aspect w.r.t. develop.

Reply:
FYI - back from shopping... I checked again using the latest commits to this branch.

Just to confirm that this change is still required for the actions to be enabled in the OnEnable / Awake / Start methods of the ProjectWideActionsSample scene when run in playmode in isolation from the rest of the sample scenes (i.e. not launched from the "Core Scene Menu" scene).

Happy to leave this for another PR in isolation however.